boards/arm/mps2_an521/mps2_an521.dts

```diff
@@ -45,6 +45,11 @@
 		reg = <0x30000000 0x1000000>;
 	};
 
+	sramx: memory@38180000 {
+		compatible = "mmio-sram";
+		reg = <0x38180000 0x8000>;
```

Comment:
strange address regarding other dts files of you pull request...Is it a specific normal that this memory is  not include in sram0?

Reply:
Yes, this seems strange, not included in the normal sram and also for all projects rather than openAMP itself.

Then, how about define the shared memory address as a macro in soc.h with the control of CONFIG_OPENAMP defined?

If using CONFIG_SHM_ADDR_xxx, that may require more prj.conf files for platfrom specific I think.

@galak

Reply:
I don't know, and probably I'm not expert enough in zephyr to determine the best solution.....
For a project integrating OpenAMP, I assume that the good practice is to define the shared memory in DTS and to resize the zephyr RAM accordingly. 
I don't know if it would be possible to create a dts overlay associated with the example project...? I couldn't find any examples.

Reply:
Sorry I'm Linux formatted, so i looked for .dtso files... overlay is possible for sample ( but maybe you already know this). There are a lot of .overlay files in samples. so you could use similar implementation. 
For instance ( code not tested) you could create a  mps2_an521_nonsecure.overlay:
&sram0: {
		reg = <0x28100000 0xF80000>;
	};

/ {
	sramx: memory@0x281F8000 {
		compatible = "mmio-sram";
		reg = <0x281F8000 0x8000>;
	};
};

This limits the RAM used by the linker script, so ensure coherency in memory allocations, and allow to declare the memory assigned to the OpenAMP  in an elegant way. So it seems like a good compromise (IMHO).

Reply:
Hi Arnaud,

Thanks for your suggestion of overlay. I updated it now.